### DISCUSSION TITLE
Input.otp How to input only numbers

### DISCUSSION THREAD

**Discussion by yichang-gst** [ant-design/ant-design#51525] [General]
Input.otp How to input only numbers

**Comment by petercat-assistant** [comment]
To ensure that an input field only accepts numbers in Ant Design, you can use the `InputNumber` component. This component is specifically designed to handle numeric input, providing a user-friendly interface for entering numbers. Here's a basic example of how to use it:

```jsx
import React from 'react';
import { InputNumber } from 'antd';

const NumericInput = () => {
  return (
    <InputNumber
      min={0} // Minimum value
      max={100} // Maximum value
      defaultValue={0} // Default value
      onChange={(value) => console.log(value)} // Handle change
    />
  );
};

export default NumericInput;
```

If you are using a regular `Input` component and want to restrict it to numbers, you can handle the input change event to filter out non-numeric characters. Here's an example:

```jsx
import React, { useState } from 'react';
import { Input } from 'antd';

const NumericInput = () => {
  const [value, setValue] = useState('');

  const handleChange = (e) => {
    const { value } = e.target;
    const numericValue = value.replace(/[^0-9]/g, ''); // Remove non-numeric characters
    setValue(numericValue);
  };

  return (
    <Input
      value={value}
      onChange={handleChange}
      placeholder="Enter numbers only"
    />
  );
};

export default NumericInput;
```

In this example, the `handleChange` function ensures that only numeric characters are retained in the input field.

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.

**Reply by yichang-gst** [reply]
I use Input.OTP

**Reply by yichang-gst** [reply]
![image](https://github.com/user-attachments/assets/f52972da-d8a6-4b8c-b421-eaac6796f6d0)
